server/handler/check_run_create_hander.go

```diff
@@ -0,0 +1,110 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/mattermost/release-bot/config"
+	"github.com/mattermost/release-bot/log"
+	"github.com/mattermost/release-bot/store"
+	"github.com/palantir/go-githubapp/githubapp"
+)
+
+type checkRunCreateHandler struct {
+	LogType           string
+	Config            *config.Config
+	ClientCreator     githubapp.ClientCreator
+	EventContextStore store.EventContextStore
+}
+
+type CheckRunCreateRequest struct {
+	Token   string `json:"token"`
+	Name    string `json:"name"`
+	Title   string `json:"title"`
+	Summary string `json:"summary"`
+}
+
+func (ccr *CheckRunCreateRequest) Log() {
+	log.Info("create_check_run").
+		Str("name", ccr.Name).
+		Str("title", ccr.Title).
+		Str("summary", ccr.Summary).
+		Msg("")
+}
+
+type CreateCheckRunResponse struct {
+	ID *int64 `json:"id"`
+}
+
+func (vh *checkRunCreateHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Info(vh.LogType).Msg("Create Check Run Request is received!")
+	if r.Method != "POST" {
+		http.Error(w, "Only POST method is supported!", http.StatusMethodNotAllowed)
+		return
+	}
+	var createCheckRunRequest CheckRunCreateRequest
+
+	err := json.NewDecoder(r.Body).Decode(&createCheckRunRequest)
+	if err != nil {
+		log.Info(vh.LogType).
+			AnErr("error", err).
+			Msg("Invalid check run request!")
+
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	createCheckRunRequest.Log()
+	eventContext, err := vh.EventContextStore.Get(createCheckRunRequest.Token)
+	if err != nil {
+		log.Info(vh.LogType).
+			AnErr("error", err).
+			Msg("Missing Event Context!")
+
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	status := "in_progress"
```

Comment:
may it's a good idea to make it enum if we have more statuses.

Reply:
may Github already has something in the golang library.

Reply:
Github does not have enum and that is the only status we need at GitHub Application.